src/components/models/Pryv.js

```diff
@@ -35,26 +37,57 @@ export type ServiceInfos = {
 }
 
 class Pryv {
-  core: (string) => string;
+  core: (string, ?string) => string;
+  init: () => null;
+  serviceInfoUrl: string;
   register: string;
+  apiUrl: string;
+
+  constructor (serviceInfoUrl: string) {
+    this.serviceInfoUrl = serviceInfoUrl;
 
-  constructor (domain: string) {
-    this.core = username => `https://${username}.${domain}`;
-    this.register = `https://reg.${domain}`;
+    this.core = function (username: string, path: ?string) {
+      if (this.apiUrl == null) {
+        console.error('apiUrl url is not set yet, call \'init()\' and wait for service info to be loaded');
+        return '';
+      }
+      path = path || '';
+      return url.resolve(this.apiUrl.replace('{username}', username), path);
+    };
+  }
+
+  async init () {
+    console.log('Pryv init');
+    let res;
+    try {
+      res = await this.getServiceInfo();
+    } catch (error) {
+      return Promise.reject(error);
+    }
+
+    this.apiUrl = res.api;
+    this.register = res.register;
+    console.log('service info fetched. api = ' + this.apiUrl + ' / reg = ' + this.register);
   }
 
   // ---------- AUTH calls ----------
 
   // GET/reg: polling with according poll key
   async poll (pollKey: string): Promise<AuthState> {
-    const res = await axios.get(`${this.register}/access/${pollKey}`);
+    if (this.register == null) {
```

Comment:
est-ce que tu peux faire ce check une seule fois dans la fonction init plutôt? profites en pour checker que tout est la (this.register, this.apiUrl)